cmd/template/dbdriver/files/service/scylla.tmpl

```diff
@@ -0,0 +1,192 @@
+func parseConsistency(cons string) (gocql.Consistency, error) {
+	switch strings.ToUpper(cons) {
+	case "ANY":
+		return gocql.Any, nil
+	case "ONE":
+		return gocql.One, nil
+	case "TWO":
+		return gocql.Two, nil
+	case "THREE":
+		return gocql.Three, nil
+	case "QUORUM":
+		return gocql.Quorum, nil
+	case "ALL":
+		return gocql.All, nil
+	case "LOCAL_ONE":
+		return gocql.LocalOne, nil
+	case "LOCAL_QUORUM":
+		return gocql.LocalQuorum, nil
+	case "EACH_QUORUM":
+		return gocql.EachQuorum, nil
+	default:
+		return gocql.Quorum, fmt.Errorf("unknown consistency level: %s", cons)
+	}
+}
+
+// Health returns the health status and statistics of the ScyllaDB cluster.
+func (s *service) Health() map[string]string {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	stats := make(map[string]string)
+
+	// Check ScyllaDB health and populate the stats map
+	stats = s.checkScyllaHealth(ctx, stats)
+
+	return stats
+}
+
+// checkScyllaHealth checks the health of the ScyllaDB cluster and adds relevant statistics.
+func (s *service) checkScyllaHealth(ctx context.Context, stats map[string]string) map[string]string {
+	startedAt := time.Now()
+
+	// Execute a simple query to check connectivity
+	query := "SELECT now() FROM system.local"
+	iter := s.Session.Query(query).WithContext(ctx).Iter()
+	var currentTime time.Time
+	if !iter.Scan(&currentTime) {
+		if err := iter.Close(); err != nil {
+			stats["status"] = "down"
+			stats["message"] = fmt.Sprintf("Failed to execute query: %v", err)
+			return stats
+		}
+	}
+	if err := iter.Close(); err != nil {
+		stats["status"] = "down"
+		stats["message"] = fmt.Sprintf("Error during query execution: %v", err)
+		return stats
+	}
+
+	// ScyllaDB is up
+	stats["status"] = "up"
+	stats["message"] = "It's healthy"
+	stats["scylla_current_time"] = currentTime.String()
+
+	// Retrieve cluster information
+	// Get keyspace information
+	getKeyspacesQuery := "SELECT keyspace_name FROM system_schema.keyspaces"
+	keyspacesIterator := s.Session.Query(getKeyspacesQuery).Iter()
+
+	stats["scylla_keyspaces"] = strconv.Itoa(keyspacesIterator.NumRows())
+	if err := keyspacesIterator.Close(); err != nil {
+		log.Fatalf("Failed to close keyspaces iterator: %v", err)
+	}
+
+	// Get cluster host information
+	var currentDatacenter string
+	clusterNodesIterator := s.Session.Query("SELECT data_center FROM system.peers").Iter()
+	clusterNodesIterator.Scan(&currentDatacenter)
+
+	// +1 is because the default connection (coordinator) is not included in the query
+	stats["scylla_cluster_size"] = strconv.Itoa(clusterNodesIterator.NumRows() + 1)
+	stats["scylla_current_datacenter"] = currentDatacenter
+	if err := clusterNodesIterator.Close(); err != nil {
+		log.Fatalf("Failed to close cluster nodes iterator: %v", err)
+	}
+
+	// Retrieve Connected Sessions
+	connectedSessionsIterator := s.Session.Query("SELECT connection_stage as connected_sessions FROM system.clients").Iter()
```

Comment:
You are going to get clients only from one host, to get all the connections you need to go over every host in the cluster.

Reply:
I'll remove this information for now. It would be a good improvement to have a default object for all drivers with some useful information like nodetool has.